src/node/resolver.ts

```diff
@@ -435,7 +436,6 @@ export function resolveOptimizedModule(
 
   const cacheDir = resolveOptimizedCacheDir(root)
   if (!cacheDir) return
-  if (!path.extname(id)) id += '.js'
   const file = path.join(cacheDir, id)
```

Comment:
1.For  `socket.io-client` the name has `.` caused not found optimize module.  
2.The `cacheKey` should end with `.js`, that will be hit cache with none `.js` files